### ISSUE TITLE
statically uninhabited schemas compile to Never with no diagnostic

### ISSUE THREAD

**Issue by tdakkota** [ogen-go/schemacompiler#39]
A schema whose constraints are impossible to satisfy compiles to `plan.NeverRepresentation` and emits **zero diagnostics**. Verified on `main` (`b494b19`):

```
{"allOf":[{"type":"object"},{"type":"null"}]}    -> NeverRepresentation, capability=direct-go-type, 0 diagnostics
{"allOf":[{"type":"string"},{"type":"number"}]}  -> NeverRepresentation, capability=direct-go-type, 0 diagnostics
```

The existing uninhabited detection (added in #13) covers only *recursive* schemas — an unguarded cycle correctly reports `SeverityError`. A non-recursive kind-disjoint intersection is detected well enough to produce `Never`, but the author is never told.

`_ref/json-schema-to-go-design.md` §15.5 makes impossible-branch elimination a normalization step, and §23 lists it under required normalization optimizations — so the compiler already knows. The gap is purely in reporting: `Never` is the correct representation, but silently handing a backend a type that can hold no value is indistinguishable from a schema that simply has no constraints.

Suggested: emit a diagnostic (probably `SeverityWarning`, or `SeverityError` where the whole root schema is uninhabited) whenever normalization produces `Never` for a schema the author wrote, with a pointer to the conflicting constraints. Care is needed not to warn for `Never` branches that normalization *legitimately* prunes inside a combinator — §15.5 pruning a dead `oneOf` branch is expected behavior, not an author error. Distinguishing "this whole schema is uninhabited" from "one branch was pruned" is the substance of the fix.

Found incidentally while reworking `nullable` (PR #30): `{allOf:[{type:object},{type:null}]}` is the natural-but-wrong way to add null via `allOf`, and it fails silently, which is what makes the reporting gap worth closing.
